xbmc/utils/URIUtils.cpp

```diff
@@ -466,7 +466,9 @@ std::string URIUtils::ChangeBasePath(const std::string &fromPath, const std::str
 CURL URIUtils::SubstitutePath(const CURL& url, bool reverse /* = false */)
 {
   const std::string pathToUrl = url.Get();
-  return CURL(SubstitutePath(pathToUrl, reverse));
+  CURL url_tmp(CURL(SubstitutePath(pathToUrl, reverse)));
+  url_tmp.SetCustomRequest(url.GetCustomRequest());
+  return url_tmp;
```

Comment:
My problem here is that the function is called "SubstitutePath" and not "SubstitutePath and Other Do Some Other Stuff Too", so that's why it doesn't belong there. By accident SubstitudePath() is called in places where you need to set the custom header right, but I wouldn't rely on the fact that this is always in the same cases as SubstitudePath. Perhaps we just need to add a new member variable to IFile ? This seems like a much cleaner approach.